modules/node_modules/@ncigdc/dux/cart.js

```diff
@@ -4,57 +4,189 @@
+    </span>
+    {notification.undo &&
+    <span style={center}>
+      <strong>
+        <i
+          className="fa fa-undo"
+          style={{
+            marginRight: '0.3rem',
+          }}
+        />
+        <a
+          style={{
+            textDecoration: 'underline',
+          }}
+          onClick={() => dispatch(toggleFilesInCart(notification.undo.files))}
+        >
+          Undo
+        </a>
+      </strong>
+    </span>
+    }
+  </NotificationColumn>,
+});
+
+function toggleFilesInCart(incomingFile) {
   return (dispatch, getState) => {
-    const incomingFilesArray = incomingFiles.constructor === Array ? incomingFiles : [incomingFiles];
+    const incomingFileArray = incomingFile.constructor === Array ? incomingFile : [incomingFile];
     const existingFiles = getState().cart.files;
-    const nextFiles = _.xorBy(existingFiles, incomingFilesArray, 'file_id');
+    const nextFiles = _.xorBy(existingFiles, incomingFileArray, 'file_id');
+
+    if (nextFiles.length > existingFiles.length) {
+      dispatch(notify(getNotificationComponent('add', `add/${incomingFile.file_name}`, {
+        action: 'Added',
+        file: incomingFile.file_name,
+        prepositon: 'to',
+        undo: {
+          files: incomingFile,
+        },
+      }, dispatch)));
+    }
+
+    if (nextFiles.length < existingFiles.length) {
+      dispatch(notify(getNotificationComponent('remove', `remove/${incomingFile.file_name}`, {
+        action: 'Removed',
+        file: incomingFile.file_name || incomingFileArray.length,
+        fileText: incomingFile.file_name ? '' : 'files',
+        prepositon: 'from',
+        undo: {
+          files: incomingFile,
+        }
+      }, dispatch)));
+    }
+
+    dispatch({
+      type: UPDATE_CART,
+      payload: nextFiles,
+    });
+  };
+}
 
-    if (nextFiles.length === existingFiles.length + 1) {
+function fetchFiles (currentFilters, total) {
+  return async dispatch => {
+    if (total <= 10000) {
+      const search = stringify({
+        filters: JSON.stringify(currentFilters),
+        size: total,
+        fields: 'access,file_id,file_size,cases.project.project_id',
+      });
+      const { data } = await fetchApi(`files?${search}`);
+      dispatch(addAllFilesInCart(data.hits));
+    } else {
       dispatch(notify({
         action: 'add',
-        id: `add/${incomingFiles.file_name}`,
         component:
-  <Column>
-    <span>Added <strong>{incomingFiles.file_name}</strong> to the cart.</span>
-    <span style={center}>
-      <UndoIcon />
-      <a onClick={() => dispatch(toggleInCart(incomingFiles))}>Undo</a>
+  <NotificationColumn>
+    <span>
+      The cart is limited to 10,000 files.
+      Please narrow down your search criteria to be able to add files to your cart.
     </span>
-  </Column>,
+  </NotificationColumn>,
       }));
     }
+  };
+}
 
-    if (nextFiles.length === existingFiles.length - 1) {
+function addAllFilesInCart(incomingFiles) {
```

Comment:
would name this to `addFilesToCart`

"FilesInCart" sounds like "InCart" describes "Files" and we're operating on Files that are already in the cart